src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -480,6 +510,53 @@ private ClusterDescription ProcessShardedChange(ClusterDescription clusterDescri
             return clusterDescription.WithServerDescription(args.NewServerDescription);
         }
 
+        private ClusterDescription ProcessStandaloneChange(ClusterDescription clusterDescription, ServerDescriptionChangedEventArgs args)
+        {
+            if (args.NewServerDescription.Type != ServerType.Standalone)
+            {
+                return RemoveServer(clusterDescription, args.NewServerDescription.EndPoint, "Server is not a standalone server.");
+            }
+
+            return clusterDescription.WithServerDescription(args.NewServerDescription);
+        }
+
+        void IDnsMonitoringCluster.ProcessDnsResults(List<DnsEndPoint> dnsEndPoints, CancellationToken cancellationToken)
```

Comment:
I'm not clear if/how this method is being tested, especially in the situation where it's called for a second time on the same instance.